### PR TITLE
ssh-key: simplified ECDSA private key decoding logic

### PR DESCRIPTION
**Pull request by tarcieri**
Followup to #351

Also includes a fix and test to ensure that short keys zero-pad MSB rather than LSB, since they're encoded as big endian.

### CODE REVIEW

**Comment by tarcieri** [comment]
cc @zaczkows

File: ssh-key/tests/private_key.rs [hunk 1]
```diff
@@ -737,7 +737,15 @@ phY3prb3dzQGwtemFjemtvd3M=
...
         }
     );
+
+    let keypair_data = key.key_data().ecdsa().unwrap();
+    assert_eq!(
+        keypair_data.private_key_bytes(),
+        hex!(
+            "000f65a3a88314a64da9c750b55639b5e532dae02163c8049b8b5a7085a9e24f8f2cfbb3ee5edf2def95f49819e83df6ce8695b24a207093972b0e5f745d03ed50d4"
+        )
+    );
```
Comment (tarcieri) [inline ssh-key/tests/private_key.rs:750]: @zaczkows it would be good if you could confirm this is the expected private key
